### PR TITLE
Support for ghc-8.8.3

### LINKED ISSUE #1670: Support for ghc-8.8.3
* Finally we have a ghc working version for windows! :smile: 
* Package managers:
  * Already available for linux via ppa (@hvr is always the first one!)
  * It still is not in homebrew: https://github.com/Homebrew/homebrew-core/pull/50677
  * Available in chocolatey but it needs maintainer corrective action: https://chocolatey.org/packages/ghc/8.8.3
  * stackage nightly still is in ghc-8.8.2: https://www.stackage.org/nightly-2020-02-27
* I am building it locally in windows and i am preparing a pr to add the new version to azure: linux-cabal and windows-cabal

### PR DESCRIPTION
**Pull request by jneira**
* It would close #1670 
* There are failing tests to be fixed before merging: https://github.com/haskell/haskell-ide-engine/issues/1670#issuecomment-593634027
* It drops ghc-8.8.1.yaml cause it has no lts

### CODE REVIEW

**Comment by alanz** [comment]
looks reasonable to me

File: stack-8.8.3.yaml [hunk 1]
```diff
@@ -1,39 +1,32 @@
...
-- lsp-test-0.10.1.0
-- hie-bios-0.4.0
-- hlint-2.2.11
+# To avoid build error with ghc-8.8.2
+# See https://github.com/haskell/haddock/issues/1123
+- git: https://github.com/haskell/haddock.git
+  commit: be8b02c4e3cffe7d45b3dad0a0f071d35a274d65
+  subdirs:
+    - haddock-api
+# - haddock-api-2.23.0
```
Comment (0xd34df00d) [inline stack-8.8.3.yaml]: Looks like `- haddock-api-2.23.1` would also work — it seems to build with ghc 8.8.2 and later. Dunno if using a released version is preferable.

Comment (jneira) [reply stack-8.8.3.yaml]: Absolutly! Thanks for pointing it out

**Comment by mouse07410** [comment]
@jneira I noticed that at least one of the failed CI jobs (https://circleci.com/gh/jneira/haskell-ide-engine/4298?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) failed in Hoogle DB build on unsuccessful attempt to retrieve **`cabal.config`** file from Stackage. That file simply doesn't exist on Stackage (at least for some GHC and/or resolver versions), like it or not.

Some time ago, trying to use `intellij-haskell` plugin (that works only with Stack), I've seen this error a lot when the plugin tried to open Cabal projects (which most of my projects are). I convinced the plugin author to just create a default `cabal.config` if the retrieval fails (as it did for me in 95% of all cases). *Finally, since I've no use of an IDE that's limited to Stack only, I dumped that plugin, and develop Haskell code only on VSCode with HIE, but that's besides the point.*

I did not examine *all* of the CI failures - but if they all break on retrieving `cabal.config`, perhaps you want to either ignore those builds, or do something similar to what we did above?

**Comment by jneira** [comment]
Thank for take a look. They are transient erros related with connectivity of ci machines and are relatively frequent, but usually they go if you run the job again.
The tests that are failing are these: https://circleci.com/gh/jneira/haskell-ide-engine/4302
I've collected then in a gist: https://gist.github.com/jneira/9bf777438bc0adbd5d6ba1e9be9350bf

**Comment by mouse07410** [comment]
> The tests that are failing are these ...

I see, thanks. Yeah, on that level I'm incompetent to help, alas.

Offhand, could you check if the problems are only with Stack-driven tests?

**Comment by fendor** [comment]
Executing the tests locally with `cabal` as the main test-driver does not yield the failing test-cases.
I changed the test setup a bit, e.g. generating a 
```yaml
cradle:
  cabal:
```
for the testdata since my stack installation has been wiped and I didnt want to run it right now, but I guess I have to :(

**Comment by jneira** [comment]
Once that @fendor has done the hard part (as usual 😸 ), fix the test suite, only left some errors in azure for windows: https://dev.azure.com/jneira/haskell-ide-engine/_build/results?buildId=744&view=results
The build has quite transient errors as well, running it again.

**Comment by jneira** [comment]
I am afraid that ghc-8.8.3 is throwing segfaults in azure ci, probably due to a ghc bug: https://gitlab.haskell.org/ghc/ghc/issues/17926
However i am able to build hie locally with ghc-8.8.3 so i am gonna remove the build from ci but i am not remove it as supported ghc version to build hie in the script install.hs for now

**Comment by fendor** [comment]
Well. Still merge?

**Comment by jneira** [comment]
@fendor i think it can be merged, i am only trying to fix azure builds, although some errors are not related with ghc-8.8.3.

**Comment by jneira** [comment]
Well, linux stack jobs have ran out of disk space since c8ed3e:
```
Filesystem      Size  Used Avail Use% Mounted on
udev            3.4G     0  3.4G   0% /dev
tmpfs           695M  8.9M  686M   2% /run
/dev/sda1        84G   80G  3.6G  96% /
tmpfs           3.4G  8.0K  3.4G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       40M   40M     0 100% /snap/hub/43
/dev/loop1       94M   94M     0 100% /snap/core/8935
/dev/sda15      105M  3.6M  101M   4% /boot/efi
/dev/sdb1        14G   35M   13G   1% /mnt
```

It make the download of ghc-8.8.1, necessary for wrapper-tests fail:

```
Installing GHC ...
/home/vsts/.stack/programs/x86_64-linux/ghc-8.8.1/lib/ghc-8.8.1/Cabal-3.0.0.0/.copyFile75307-1209.tmp: copyFile: resource exhausted (No space left on device)
make[1]: *** [install_packages] Error 1
make: *** [install] Error 2
Run from: /home/vsts/.stack/programs/x86_64-linux/ghc-8.8.1.temp/ghc-8.8.1/


Error: Error encountered while installing GHC with
         make install
         run in /home/vsts/.stack/programs/x86_64-linux/ghc-8.8.1.temp/ghc-8.8.1/
       
       The following directories may now contain files, but won't be used by stack:
         - /home/vsts/.stack/programs/x86_64-linux/ghc-8.8.1.temp/
         - /home/vsts/.stack/programs/x86_64-linux/ghc-8.8.1/
       
       For more information consider rerunning with --verbose flag
       

Command `stack ghc -- --numeric-version` failed.

version checking
  picks up a stack.yaml with 8.8.1 FAILED [1]
  picks up a stack.yaml with 8.6.5
  picks up whatever version of ghc is on this machine

Failures:

  test/wrapper/HieWrapper.hs:18:9: 
  1) version checking picks up a stack.yaml with 8.8.1
       expected: "8.8.1"
        but got: "8.8.3"
```

It is not related with this pr, so i would merge this if other builds are succesful.

**Comment by jneira** [comment]
Almost all building jobs are due to #1729 so i think it can be merged

**Pull request merged**